core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs

```diff
@@ -10,34 +37,74 @@ namespace Test.Microsoft.Identity.Core.UIAutomation
 {
     public class CoreMobileTestHelper
     {
+        public static bool PlatformParameters = false;
+
         public static void PerformSignInFlow(ITestController controller, IUser user)
         {
-            string passwordInputID = string.Empty;
-            string signInButtonID = string.Empty;
-
-            if (user.IsFederated)
+            if (PlatformParameters)
             {
-                switch (user.FederationProvider)
-                {
-                    case FederationProvider.AdfsV3:
-                    case FederationProvider.AdfsV4:
-                        passwordInputID = CoreUiTestConstants.AdfsV4WebPasswordID;
-                        signInButtonID = CoreUiTestConstants.AdfsV4WebSubmitID;
-                        break;
-                    default:
-                        passwordInputID = CoreUiTestConstants.WebPasswordID;
-                        signInButtonID = CoreUiTestConstants.WebSubmitID;
-                        break;
-                }
+                PerformSignInFlowWithPromptBehaviorAlways(controller, user);
             }
+
```

Comment:
your re-use of logic could be made even simpler. Consider calling the PerformSignInFlowWith...Always from the external test. then there is no parameter you have to set (and thus class state). Also everything regarding getting the identifiers for the username/password fields can be extracted. I think if you consider the first suggestion you will identify a more optimal way of structuring the code.

Reply:
cool. thanks for the suggestion. ☃️ 😼